classes/classes/Scenes/Areas/GlacialRift/Amarok.as

```diff
@@ -0,0 +1,164 @@
+// By Foxwells
+// Amarok, a big ass wolf from Inuit mythology. Idek. Glacial Rift is depressingly empty
+// Since it's a big ass god wolf, it's pretty hard to beat
+// Herm, since there's no set gender for it in its mythology
+// Comes with a unique bad end if you lose to it too many times, kinda like the Phoenix
+// Also: Wolf TF! :D Because like, really, why hasn't that been implemented yet? It could be sooo coool <3
+
+package classes.Scenes.Areas.GlacialRift 
+{
+	import classes.*;
+	import classes.internals.WeightedDrop;
+	
+	public class Amarok extends Monster
+	{
+		
+		protected function amarokClaw():void {
+			//Blind
+			if(findStatusEffect(StatusEffects.Blind) >= 0) {
+				outputText("The Amarok lunges for you, attempting to slash you with one of its paws, but misses completely due to its blindness.", false);
+				combatRoundOver();
+				return;
+			}
+			//Too many things for evasion...
+			if(player.findPerk(PerkLib.Evade) >= 0 && player.spe >= 35 && rand(3) != 0) {
+				outputText("The Amarok lunges for you, paw ready to strike, but you easily roll out of the way thanks to your talents in evasion.", false);
+			}
+			else if(player.findPerk(PerkLib.Misdirection) >= 0 && rand(100) < 20 && player.armorName == "red, high-society bodysuit") {
+				outputText("Using Raphael's teachings and the movement afforded by your bodysuit, you anticipate and sidestep the Amarok's attempt to claw you.", false);
+			}
+			else if(player.findPerk(PerkLib.Flexibility) >= 0 && player.spe > 30 && rand(10) != 0) {
+				outputText("The Amarok charges at you, attempting to claw you with one of its massive paws. You dodge, using your incredible flexibility to twist out of the way.", false);
+			}
+			else if (player.getEvasionRoll()) {
+				outputText("The Amarok throws itself at you, attempting to slash you with its claws. Luckily, you manage to leap out of the way.", false);
+				combatRoundOver();
+				return;
+			}
+			else {
+				//Damage roll
+				outputText("The Amarok throws itself at you and rakes one of its hefty paws across you. Its claws slice you open and draw blood.", false);
+				var damage:int = ((str + 50) + rand(100));
+				damage = player.reduceDamage(damage);
+				player.takeDamage(damage, true);
+				if(player.HP <= 0) {
+					doNext(game.combat.endHpLoss);
+					return;
+				}
+			}
+			combatRoundOver();
+		}
+		protected function amarokTail():void {
+			//AMAROK used TAIL SLAP!
+			outputText("The Amarok rushes up to you and immediately turns heel, attempting to crash its tail into you. ", false);
+			//Blind check...
+			if(findStatusEffect(StatusEffects.Blind) >= 0) {
+				outputText("Luckily, though, its blindness causes it to misjudge your location and it misses entirely.", false);
+				combatRoundOver();
+				return;
+			}
+			//Evasion stuffs
```

Comment:
You should either use getEvasionRoll (returns unspecified boolean) or getEvasionReason (returns which reason fired to make flavor text). This way you'll check evasion twice - first in your code, then in getEvasionRoll.